### PR TITLE
PR: IN Profile Listings Page

### PR DESCRIPTION
**Pull request by ababber**
### What this PR does (required):
- updated profile listings page with backend get route to load multiple users based on isSitter property in db, integrated FE and BE so that profile listings page loads data from db

### Screenshots / Videos (front-end only):
https://user-images.githubusercontent.com/18492009/145140763-982808b5-59df-4f8a-b62e-927b3bb8ff59.mov

### Any information needed to test this feature (required):
- team-gingerbread s3 bucket access required, load app, add users with isSitter set to true and upload pictures, navigate to profile listings page, and the new users created will be loaded to profile listings page

### Any issues with the current functionality (optional):
- profile listings page is not integrated with landing page and does not load based on address
- cannot load based on availability because feature is not complete

### CODE REVIEW

File: server/controllers/deleteImage.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  await Profile.findByIdAndUpdate(String(_id), { photo: { url: "", key: "" } }, (err, result) => {
+    if (err) {
+      res.status(500).send(`A problem occurred while deleting an image. ${err}`);
+    } else res.status(200).send(`Image deleted. ${result}`);
+  });
+};
+
+// @route DELETE /image/delete?
+// @desc delete profile photo in S3 and DB
+// @access Private
```
Comment (ezeilo-su) [inline server/controllers/deleteImage.js:44]: :+1:

File: server/controllers/uploadImage.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    } else {
+      cb(null, false);
+      return cb(new Error("Please upload a jpeg or png file type."));
+    }
+  },
+}).array("images");
+
+// @route POST /image/upload
+// @desc upload profile photo to S3 and write link to DB
+// @access Private
```
Comment (ezeilo-su) [inline server/controllers/uploadImage.js:66]: :+1:

File: client/src/pages/ProfileListings/ProfileListings.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      <Grid item>
+        <Grid container spacing={6} className={classes.profilePreviewWrapper}>
+          {!profileData ? null : renderedPreviews}
+        </Grid>
+      </Grid>
+      <Grid item spacing={1}>
+        <Button
+          size="large"
+          variant="outlined"
+          onClick={() => console.log('load more users')}
```
Comment (ezeilo-su) [inline client/src/pages/ProfileListings/ProfileListings.tsx:85]: Please let's get rid of this console log

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Nice integration. Just left a few comments for you to have a look. Please also resolve the merge conflict in this branch.

File: server/controllers/updateIsSitter.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  });
+  return response;
+};
+
+const changeIsSitter = async (_id, value, res) => {
+  await Profile.findByIdAndUpdate(String(_id), { isSitter: value }, (err, result) => {
+    if (err) {
+      res.status(500).send(`A problem occurred while changing sitter status. ${err}`);
+    } else res.status(200).send(`Sitter status changed. ${result}`);
+  });
```
[the remark this answers is not in the document]
Comment (ababber) [reply server/controllers/updateIsSitter.js:16]: Changed in the following commit: https://github.com/hatchways/team-gingerbread/pull/93/commits/f1ad6780a1fdf1a99667d7729fc694ad58526f1a

File: server/controllers/updateIsSitter.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+// @route PATCH /profile/sitter
+// @desc become a sitter or not
+// @access Private
+exports.updateIsSitter = async (req, res) => {
+  const { _id } = req.query;
```
[the remark this answers is not in the document]
Comment (ababber) [reply server/controllers/updateIsSitter.js]: Fixed in the following commit: https://github.com/hatchways/team-gingerbread/pull/90/commits/eba2d2f159c0699b085c3921a500fe1be51ed898

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx [hunk 6]
```diff
@@ -94,13 +100,17 @@ export default function EditProfileTab(): JSX.Element {
         {accountType === 'partner' && (
           <Box className={classes.section}>
             <FormLabel>
-              <Typography className={classes.label}>i&apos;m available</Typography>
+              {isSitter ? (
+                <Typography className={classes.label}>i&apos;m a sitter</Typography>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:104]:
```suggestion
                <Typography className={classes.label}>
                  {isSitter ? "i'm a sitter" : "i'm not a sitter"}
                 </Typography>
```

File: client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx [hunk 7]
```diff
@@ -3,24 +3,46 @@ import { Typography, Box, Grid, Avatar, Button, InputLabel } from '@material-ui/
...
     const fileURL = file ? URL.createObjectURL(file) : '';
+    const images = [];
     if (file && (file.type === 'image/jpeg' || file.type === 'image/png' || file.type === 'image/jpg')) {
       setIsUploaded(true);
       setIsWrongFileType(false);
       setDisplayPhoto(fileURL);
+      if (loggedInUser) {
+        images.push(file);
+        uploadImage(images, loggedInUser.profile).then((data) => {
+          if (data.error) updateSnackBarMessage('An error occurred while uploading the image.');
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfilePhotoTab/ProfilePhotoTab.tsx:31]: Good work making these small changes but resort to making them in a separate PR :)

File: client/src/pages/ProfileListings/ProfilePreview/ProfilePreview.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  location: string;
+  payRate: number;
+}
+
+const ProfilePreview = ({ img, name, subtitle, rating, description, location, payRate }: PreviewProps): JSX.Element => {
+  const classes = useStyles();
+
+  return (
+    <Card className={classes.root}>
+      <CardContent className={classes.content}>
```
Comment (rajivtitus) [inline client/src/pages/ProfileListings/ProfilePreview/ProfilePreview.tsx:22]:
```suggestion
      <CardActionArea comopnent={Link} to=`/profile-details/${profile.id}`
      <CardContent className={classes.content}>
```
Adding the cardaction area here to navigate from the listing page to the details page would a nice tough 👍🏽

File: server/controllers/profile.js [hunk 9]
```diff
@@ -32,9 +24,20 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
...
-    },
-  });
+  res.status(200).json({ success: { profile } });
 });
+
+// @route GET /profile/listings
+// @desc load users from db
+// @access Public
+exports.populateListings = async (req, res) => {
+  const { numOfUsers } = req.query;
```
Comment (rajivtitus) [inline server/controllers/profile.js:34]: Good work! Additionally, you can have pagination with page numbers/ profiles per page :)

A video I found on the internet: https://www.youtube.com/watch?v=soWg_UtD_AM

**Review by rajivtitus — APPROVED** [review]
Great work Ankit! Just a couple of minor comments for you to address before merging 👍🏽

**Pull request merged**